llvm/include/llvm/CodeGen/GlobalISel/GenericMachineInstrs.h

```diff
@@ -240,13 +240,22 @@ class GMergeLikeInstr : public GenericMachineInstr {
     case TargetOpcode::G_MERGE_VALUES:
     case TargetOpcode::G_CONCAT_VECTORS:
     case TargetOpcode::G_BUILD_VECTOR:
+    case TargetOpcode::G_SHUFFLE_VECTOR:
       return true;
     default:
       return false;
     }
   }
 };
 
+/// Represents a G_SHUFFLE_VECTOR
+class GShuffleVector : public GMergeLikeInstr {
```

Comment:
G_SHUFFLE_VECTOR takes the shufflemask as the last operand, which shouldn't be part of the `getNumSources` I think?
To me that's what makes G_SHUFFLE_VECTOR different compared to the other "merge like" instructions.
I'd rather not make it part of that group, as this can have surprising effects to other existing combines

Reply:
I can see that argument. I was looking at it from the perspective of what the result of the function would be. I'll separate them, thanks.